frontend/src/components/wallet/BalanceCard.tsx

```diff
@@ -0,0 +1,66 @@
+import { ArrowUp } from "lucide-react";
+import { Link } from "react-router-dom";
+import { Button } from "src/components/ui/button";
+import {
+  Card,
+  CardContent,
+  CardFooter,
+  CardHeader,
+  CardTitle,
+} from "src/components/ui/card";
+import { BalancesResponse } from "src/types";
+
+type BalanceCardProps = {
+  balances: BalancesResponse;
+  hasChannelManagement: boolean;
+  isSpending?: boolean;
+};
+
+function BalanceCard({
```

Comment:
This was unfortunately a poor refactoring and introduced at least 2 bugs, and now requires a lot of conditionals.

You could pass args to the component rather than using conditionals, and you didn't even add them all (for example the arrow icon is not right for receive, now the receiving card is shown for LN clients that do not support channel management, ...

and this is only what I spotted from looking at the code, there may be more

Reply:
> now the receiving card is shown for LN clients that do not support channel management

I thought it would be useful to see the capacity and show the increase button only if there is channel management like we do in the Send card. Is there any reason why we show the spending card without the button and don't show it at all for receiving card?

And sorry, I missed the icon, let me know if the above is okay and depending on that I'll revert the component or add the icon (if it's okay to have another conditional)

Reply:
Let's chat about this one on a call